src/Repository/ReferentRepository.php

```diff
@@ -29,14 +30,37 @@ public function findByStatus(string $status = Referent::ENABLED): array
     }
 
     public function findOneByEmailAndSelectPersonOrgaChart(string $email): ?Referent
+    {
+        return $this->createQueryBuilderWithEmail($email)
+            ->addSelect('referent_person_links')
+            ->getQuery()
+            ->getSingleResult()
+        ;
+    }
+
+    public function findOneByEmail(string $email): Referent
+    {
+        return $this->createQueryBuilderWithEmail($email)
+            ->getQuery()
+            ->getSingleResult()
+        ;
+    }
+
+    public function findIfIsRoot(string $email): bool
```

Comment:
The purpose of a repository is to find/save stuff/entity. This one is business logic.
You fixed the semantic issue with the voter so now you can use any method you want in this voter. So let's remove this method or give another name  ?